acoular/tbeamform.py

```diff
@@ -102,141 +100,23 @@ class BeamformerTime(TimeOut):
-    # Now governed by :attr:`steer` trait.
-    c = Property()
-
-    def _get_c(self):
-        return self._steer_obj.env.c
-
-    def _set_c(self, c):
-        msg = (
-            "Deprecated use of 'c' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector'. The 'c' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        self._steer_obj.env.c = c
-
-    # :class:`~acoular.grids.Grid`-derived object that provides the grid locations.
-    # Deprecated! Only kept for backwards compatibility.
-    # Now governed by :attr:`steer` trait.
-    grid = Property()
-
-    def _get_grid(self):
-        return self._steer_obj.grid
-
-    def _set_grid(self, grid):
-        msg = (
-            "Deprecated use of 'grid' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector'. The 'grid' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        self._steer_obj.grid = grid
-
-    # :class:`~acoular.microphones.MicGeom` object that provides the microphone locations.
-    # Deprecated! Only kept for backwards compatibility.
-    # Now governed by :attr:`steer` trait
-    mpos = Property()
-
-    def _get_mpos(self):
-        return self._steer_obj.mics
-
-    def _set_mpos(self, mpos):
-        msg = (
-            "Deprecated use of 'mpos' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector' which has an attribute 'mics'. The 'mpos' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        self._steer_obj.mics = mpos
-
-    # Sound travel distances from microphone array center to grid points (r0)
-    # and all array mics to grid points (rm). Readonly.
-    # Deprecated! Only kept for backwards compatibility.
-    # Now governed by :attr:`steer` trait
-    r0 = Property()
-
-    def _get_r0(self):
-        msg = (
-            "Deprecated use of 'r0' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector'. The 'r0' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        return self._steer_obj.r0
-
-    rm = Property()
-
-    def _get_rm(self):
-        msg = (
-            "Deprecated use of 'rm' trait. Please use the 'steer' trait with an object of class  "
-            "'SteeringVector'. The 'rm' trait will be removed in version 25.01."
-        )
-        warn(msg, DeprecationWarning, stacklevel=2)
-        return self._steer_obj.rm
-
-    # --- End of backwards compatibility traits --------------------------------------
-
+    steer = Instance(SteeringVector(), SteeringVector)
     #: Number of channels in output (=number of grid points).
-    numchannels = Delegate('grid', 'size')
+    numchannels = Property(depends_on=['steer.grid'])
```

Comment:
This line does not work as expected. Numchannels only changes when the steer.grid class changes, but not if the grid size changes.

Reply:
This line is still included.